submodules/islandora_webform_ingest/includes/utilities.inc

```diff
@@ -496,40 +503,35 @@ function iwi_set_form_element_byhash(&$form, &$form_state, &$cloned_elements, $e
             case "tabpanel":
               $tab = $element->findElement($child_element_hash);
               if ($tab) {
-                if (!array_key_exists($child_element_hash, $cloned_elements)) {
-                  // If not already cloned clone it.
-                  $new_tab = clone $tab;
-
-                  $set_element_properties = function ($element) {
-                    $element->default_value = NULL;
-                  };
-                  // Set element properties on the descendants.
-                  $new_tab->eachDecendant($set_element_properties);
-
-                  $element->adopt($new_tab);
-                  $new_tab_drupal = $new_tab->toArray();
-
-                  drupal_array_set_nested_value($new_tab_drupal, array_merge($future_partial_path, array('#attributes')), $input_value['#attributes']);
-                  drupal_array_set_nested_value($new_tab_drupal, array_merge($future_partial_path, array('#default_value')), $input_value['#default_value']);
-
-                  $special_element_ref[] = $new_tab_drupal;
-
-                  // Appending the drupal form array back to where it belongs.
-                  // Lets store where do we put this one, so we can later just
-                  // fill out the values if we need to.
-                  $special_element_children = element_children($special_element_ref);
-                  $cloned_elements[$child_element_hash] = end($special_element_children);
+                $special_element_children = element_children($special_element_ref);
+                if (!isset($special_element_children[$offset])) {
+                  $lastkid = end($special_element_children);
+                  for ($i = $lastkid; $i < $offset; $i++) {
```

Comment:
Getting an infinite loop here because $lastkid is not numeric. I was surprised to find that php is happy to increment string variables:
> php > $i = 'dog';
> php > $i++;
> php > echo $i;
> doh

I'm confused as to how this is working for you. You mentioned that '[0]' needs to be part of the path. Maybe that is why this is failing. But the paths created for the iwi field mapping do not include these (see submodules/islandora_webform_ingest/includes/utilities.inc:216). Have you made a change to the iwi_recursive_form_parsing() function?